microprofile/reactive-streams/src/main/java/io/helidon/microprofile/reactive/HelidonReactiveStreamsEngine.java

```diff
@@ -526,18 +527,32 @@ static Object build(Iterable<Stage> graph, Mode mode) throws UnsupportedStageExc
     }
 
     static void complete(CompletableFuture<Object> cf) {
-        ForkJoinPool.commonPool().submit(() -> {
+        coupledExecutor.execute(() -> {
             cf.complete(null);
-            return null;
         });
     }
 
     static void fail(CompletableFuture<Object> cf, Throwable ex) {
-        ForkJoinPool.commonPool().submit(() -> {
+        coupledExecutor.execute(() -> {
             cf.completeExceptionally(ex);
-            return null;
         });
     }
+
+    // Workaround for a TCK bug when calling cancel() from any method named onComplete().
+    private static volatile ExecutorService coupledExecutor = ForkJoinPool.commonPool();
```

Comment:
For future reference TCK fix: https://github.com/reactive-streams/reactive-streams-jvm/pull/483